docs/field_write_semantics.adoc

```diff
@@ -0,0 +1,129 @@
+
+== Field Write Semantics
+
+Each field in a register is parameterized with a position -- the closed
+(inclusive) range from LSB to MSB, from which a mask can be derived -- and a
+_write function_.
```

Comment:
Suggestion: I think "write function" is the most accurate mathematical term for this...but something like "access modifier" may be a more accessible term for users.

You could go on to explain that the access modifiers are modeled as write functions and carry on with the explanation below.

Reply:
Sure - at the moment I just wanted to write down this theory. I reality users don't really need to know about this stuff; we  can totally present the options as just a few of the conventionally-named possible values (eg read_only).